protocols/stream/src/control.rs

```diff
@@ -103,6 +103,15 @@ impl fmt::Display for OpenStreamError {
     }
 }
 
+impl std::error::Error for OpenStreamError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            Self::Io(error) => Some(error),
```

Comment:
[question]

Silly question, do we also need to handle the UnsupportedProtocol variant?

Reply:
`StreamProtocol` does not implement `Error` so we could simplify ignore it. If a future change that would add something that does, we could then add it to `OpenStreamError::source`

Reply:
Makes sense. If we can't fully convert OpenStreamError to error, is this losy conversion safe for error handling?

(asking to learn :smile:)